sonaranalyzer-dotnet/src/SonarAnalyzer.CSharp/Helpers/CollectionUtils.cs

```diff
@@ -82,6 +82,39 @@ public static IEnumerable<T> WhereNotNull<T>(this IEnumerable<T> enumerable)
             where T : class
         {
             return enumerable.Where(e => e != null);
+        }
+
+        /// <summary>
+        /// Applies a specified function to the corresponding elements of two sequences,
+        /// producing a sequence of the results. If the collections have different length
+        /// default(T) will be passed in the operation function for the corresponding items that 
+        /// do not exist.
+        /// </summary>
+        /// <typeparam name="TFirst">The type of the elements of the first input sequence.</typeparam>
+        /// <typeparam name="TSecond">The type of the elements of the second input sequence.</typeparam>
+        /// <typeparam name="TResult">The type of the elements of the result sequence.</typeparam>
+        /// <param name="first">The first sequence to merge.</param>
+        /// <param name="second">The second sequence to merge.</param>
+        /// <param name="operation">A function that specifies how to merge the elements from the two sequences.</param>
+        /// <returns>An System.Collections.Generic.IEnumerable`1 that contains merged elements of
+        /// two input sequences.</returns>
+        public static IEnumerable<TResult> Merge<TFirst, TSecond, TResult>(this IEnumerable<TFirst> first,

```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Reduce the number of generic parameters in the 'CollectionUtils.Merge' method to no more than the 2 authorized. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=csharpsquid%3AS2436)